rllib/core/rl_module/torch/torch_rl_module.py

```diff
@@ -1,25 +1,135 @@
+    """Configuration options for RLlib's usage of torch.compile in RLModules.
+
+    # On `torch.compile` in Torch RLModules
+    `torch.compile` invokes torch's dynamo JIT compiler that can potentially bring
+    speedups to RL Module's forward methods.
+    This is a performance optimization that should be disabled for debugging.
+
+    General usage:
+    - Usually, you only want to `RLModule._forward_train` to be compiled on
+      instances of RLModule used for learning. (e.g. the learner)
+    - In some cases, it can bring speedups to also compile
+      `RLModule._forward_exploration` on instances used for exploration. (e.g.
+      RolloutWorker)
+    - In some cases, it can bring speedups to also compile
+      `RLModule._forward_inference` on instances used for inference. (e.g.
+      RolloutWorker)
+
+    Note that different backends are available on different platforms.
+    Also note that the default backend for torch dynamo is "aot_eager" on macOS.
+    This is a debugging backend that is expected not to improve performance because
+    the inductor backend is not supported on OSX so far.
+
+    Args:
+        compile_forward_train: Whether to compile the forward_train method.
+        compile_forward_inference: Whether to compile the forward_inference method.
+        compile_forward_exploration: Whether to compile the forward_exploration method.
+        torch_dynamo_backend: The torch.dynamo backend to use.
+        kwargs: Additional keyword arguments to pass to `torch.compile()`
+
+    """
+
+    compile_forward_train: bool = False
+    compile_forward_inference: bool = False
+    compile_forward_exploration: bool = False
+    torch_dynamo_backend: str = "aot_eager" if sys.platform == "darwin" else "inductor"
+    torch_dynamo_mode: str = "reduce-overhead"
+    kwargs: dict = field(default_factory=lambda: dict())
+
+
+def compile_wrapper(rl_module: "TorchRLModule", compile_config: TorchCompileConfig):
+    """A wrapper that compiles the forward methods of a TorchRLModule.
+
+    Possibly replaces the forward methods with compiled versions, depending on
+    compile_config. Note that this also makes it so that exploration and inference
+    do not trace grads because this can lead to memory leaks and slowness.
+    """
+
+    # TODO(Artur): Remove this once our requirements enforce torch >= 2.0.0
+    if int(torch.__version__[0]) < 2 and (
+        compile_config.compile_forward_train
+        or compile_config.compile_forward_inference
+        or compile_config.compile_forward_exploration
+    ):
+        raise ValueError("torch.compile is only supported from torch 2.0.0")
+
+    if compile_config.compile_forward_train:
+        compiled_forward_train = torch.compile(
+            rl_module._forward_train,
+            backend=compile_config.torch_dynamo_backend,
+            mode=compile_config.torch_dynamo_mode,
+            **compile_config.kwargs
+        )
+
+        rl_module._forward_train = compiled_forward_train
```

Comment:
I have added a test. A similar test is hard to do for CPU memory because that is shared with other objects and thus less accurate to track. I ran the resulting GPU test I wrote on a cluster and it passes.